catalyst_voices/packages/internal/catalyst_voices_repositories/lib/src/database/dao/documents_v2_dao.dart

```diff
@@ -9,60 +9,128 @@ import 'package:collection/collection.dart';
 import 'package:drift/drift.dart';
 
 abstract interface class DocumentsV2Dao {
-  /// Returns the total number of documents in the table.
-  Future<int> count();
+  /// Counts the number of documents matching the provided filters.
+  ///
+  /// [type] filters by the document type (e.g., proposal, comment).
+  /// [ref] filters by the document's own identity.
+  ///   - If [DocumentRef.isExact], counts matches for that specific version.
+  ///   - If [DocumentRef.isLoose], counts all versions of that document ID.
+  /// [refTo] filters documents that *reference* the given target.
+  ///   - Example: Count all comments ([type]=comment) that point to proposal X ([refTo]=X).
+  Future<int> count({
+    DocumentType? type,
+    DocumentRef? ref,
+    DocumentRef? refTo,
+  });
+
+  /// Deletes documents from the database, preserving those with types in [notInType].
+  ///
+  /// If [notInType] is null or empty, this may delete *all* documents (implementation dependent).
+  /// Typically used for cache invalidation or cleaning up old data while keeping
+  /// certain important types (e.g. keeping local drafts or templates).
+  ///
+  /// Returns the number of deleted rows.
+  Future<int> deleteWhere({
+    List<DocumentType>? notInType,
```

Comment:
```suggestion
    List<DocumentType>? typeNotIn,
```

Suggested one reads:
```delete where type not in ...```

Current one reads:
```delete where not in type ...```